packages/ui/src/pages/JobPage/JobPage.tsx

```diff
@@ -0,0 +1,123 @@
+import { useParams, useHistory, useLocation } from 'react-router-dom';
+import { AppJob, AppQueue, Status } from '@bull-board/api/typings/app';
+import React, { useEffect, useRef, useState } from 'react';
+import { Store } from '../../hooks/useStore';
+import s from '../QueuePage/QueuePage.module.css';
+import { JobCard } from '../../components/JobCard/JobCard';
+import { ArrowLeftIcon } from '../../components/Icons/ArrowLeft';
+import { Button } from '../../components/JobCard/Button/Button';
+
+export const JobPage = ({
+  actions,
+  queue,
+  selectedStatus,
+}: {
+  queue: AppQueue | null;
+  actions: Store['actions'];
+  selectedStatus: Store['selectedStatuses'];
+}) => {
+  const { search } = useLocation();
+  const { name, jobId } = useParams<any>();
+  const [job, setJob] = useState<AppJob>();
+  const [jobState, setJobState] = useState<Status>(selectedStatus[queue?.name || '']);
+  const intervalRef = useRef<NodeJS.Timeout | null>();
+  const history = useHistory();
+
+  const fetchJob = async () => {
+    const jobResponse = await actions.getJob(name)(jobId)();
+    setJob(jobResponse[jobId]);
+  };
+
+  const pollForJob = () => {
+    if (!!intervalRef.current) return;
+    fetchJob();
+
+    intervalRef.current = setInterval(() => {
+      fetchJob();
+    }, 5000);
+  };
+
+  const retryJob = async () => {
+    if (!queue || !job) return;
+
+    await actions.retryJob(queue?.name, 'failed')(job)();
+    setJobState('active');
+    pollForJob();
+  };
+
+  const cleanJob = async () => {
+    if (!queue || !job) return;
+
+    await actions.cleanJob(queue?.name)(job)();
+    history.push(`/queue/${queue.name}`);
+  };
+
+  useEffect(() => {
+    if (!queue || !job) return;
+
+    if (jobState === 'active') {
+      // transition from active to completed
+      if (job?.progress === 100 && !job.failedReason) {
```

Comment:
I haven't been able to figure out how to interrogate a job to see which state queue it's in at any given time. If we had that info much of this complexity could go away 🔥